### ISSUE TITLE
Middlewares are not invoked in the order they are added

### ISSUE THREAD

**Issue by cosic196** [dotnet/command-line-api#1868]
As the title suggests, the middlewares are not running in the same order they are added (using the same `MiddlewareOrder` for each middleware).

Not sure if this is by design, but I was expecting that adding middlewares in a particular order to the command line builder would invoke them in the order I added them (like the ASP.NET middlewares).

Is there a reason for not keeping the same order when invoking middlewares?

**Comment by cosic196** [comment]
Thanks @elgonzo !

Coming from ASP.NET middlewares, it would make sense to me that they are executed in the same order as they were added. Instead, to achieve this, I would need a "super" middleware that does the correct sorting, which seems unnecessarily complicated.

Maybe it would've been better if I rephrased it as "_System.Commandline middlewares with the same priority aren't executed in the same order as they were added_" and then pointed out the `UseExceptionHandler(...)` as an example for reproducing the "issue".

If this is the intended behavior, I would still like to know if it's possible to change this behavior to actually execute middlewares in the order of addition. I can make a PR, but obviously this suggestion needs to be considered by the community first.

**Comment by elgonzo** [comment]
I tend to agree with your view. On an intuitive level, it makes completely sense to expect the sequence in which middlewares with the same priority are executed is the same in which they were added to the CommandlineBuilder.

I can't speak for the maintainers of the project, but it shouldn't be too difficult to achieve stable sorting. Either by switching over to Linq's _OrderBy_ (which is doing stable sorting as parts of its contract), or - if _OrderBy_ is undesired, using a three-element tuple `(InvocationMiddleware, Order, SequenceIndex)` for \_middlewareList, with the comparator then acting on both the given order and sequence index (basically a unique number for each middleware that is incremented for each middleware added to \_middlewareList) to achieve stable sorting.

**Comment by KalleOlaviNiemitalo** [comment]
The middleware API has been deleted in 2.0.0-beta5 so this issue is no longer relevant.
